File: programming/project-learning/lua/algorithms/hashing.lua
-- Seed the random number generator with the current time
math.randomseed(os.time())

-- Prompt the user to enter the size of the data
print("Enter data size: ")
local dataSize = tonumber(io.read()) -- Read the user input and convert it to a number

local data = {} -- Table to store generated data
local hashTable = {} -- Table to represent the hash table

-- Loop to generate random data
for i = 1, dataSize do
    table.insert(hashTable, nil) -- Initialize the hash table with nil values
    local nameLength = math.random(3, 12) -- Generate a random length for the string (between 3 and 12)
    local tempString = "" -- Temporary string to build the random name
    for j = 1, nameLength do
        local char = string.char(math.random(65, 90)) -- Generate a random uppercase letter (ASCII 65-90)
        tempString = tempString .. char -- Append the character to the temporary string
    end
    table.insert(data, tempString) -- Insert the generated string into the data table
end

-- Loop to print the generated data
for i, v in ipairs(data) do
    print(v) -- Print each string in the data table
end

File: programming/project-learning/r/algorithms/hashing.r
set.seed(Sys.time())  # Set the seed for random number generation based on the current time

cat("Enter data size: ")  # Prompt the user to enter the size of the data
dataSize <- as.integer(readLines("stdin", n = 1))  # Read the input from the user and convert it to an integer

data <- character(dataSize)  # Initialize a character vector to store the data
hashTable <- vector("list", dataSize)  # Initialize a list to store the hash table

for (i in 1:dataSize) {  # Loop through the range from 1 to dataSize
    nameLength <- sample(3:12, 1)  # Randomly select a name length between 3 and 12
    tempString <- paste0(sample(LETTERS, nameLength, replace = TRUE), collapse = "")  # Generate a random string of the selected length
    data[i] <- tempString  # Store the generated string in the data vector
}

print(data)  # Print the generated data